### PR TITLE
Experimental shadowing output

### PR DESCRIPTION
**Pull request by Drapegnik**
Possible error output for #5 

screen shot 2018-05-27 at 03 11 01

### CODE REVIEW

**Comment by DaQuirm** [comment]
looks great (both the code and the output 😺), we could use a library for colorized output, [rainbow](http://hackage.haskell.org/package/rainbow) for example

**Comment by DaQuirm** [comment]
Let's leave this one like this and then improve it with monads 😸

File: test/ShadowingTest.hs [hunk 1]
```diff
@@ -14,11 +14,11 @@ tests =
     , testCase "Abs with underscore" $
         checkShadowing ["_x"] abs_with_underscore @?= []
     , testCase "Abs twice shadow" $
-        checkShadowing [] abs_twice_shadow @?= ["y", "x"]
+        checkShadowing [] abs_twice_shadow @?= [(ShadowVar "y" lit), (ShadowVar "x" lit)]
```
Comment (Drapegnik) [inline test/ShadowingTest.hs]: @DaQuirm, here I need only `Identifier` from `ShadowVar`. How can I beautifully unpack it here & compare?

Comment (DaQuirm) [reply test/ShadowingTest.hs]: You could write a function: `ShadowVar -> identifier` and `map` it. Or you could make `ShadowVar` a record and get this function for free.

**Pull request merged**